### PR TITLE
fix(SSO): test login protocal not match

### PR DESCRIPTION
**Pull request by ungreat**
#### What type of PR is this?
fix test login redirect url schema may not match. see https://github.com/oceanbase/odc/issues/210

#### What this PR does / why we need it:
ODC not use {baseUrl} to rewrite redirect Url, instead of , use the host obtained by the front-end.

#### Which issue(s) this PR fixes:

Fixes #https://github.com/oceanbase/odc/issues/210

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/model/Oauth2Parameter.java [hunk 1]
```diff
@@ -100,6 +99,11 @@ private Provider toProvider(String issueUrl) {
         return provider;
     }
 
+    @SneakyThrows
+    private String getRedirectHost(String redirectUrl) {
+        URL url = new URL(redirectUrl);
+        return url.getProtocol() + "://" + url.getHost() + (url.getPort() > 0 ? ":" + url.getPort() : "");
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/model/Oauth2Parameter.java:105]: if port is 80 for http protocal, or port is 443 for https protocal, skip  port part in url.

Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/integration/model/Oauth2Parameter.java:105]: may refer spring util `UrlUtils.buildFullRequestUrl`

Comment (ungreat) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/integration/model/Oauth2Parameter.java]: `UrlUtils.buildFullRequestUrl` is more applicable to Request objects, which returns the default port of the protocol when you get port. Here use java.net.URL to parse url string, if there is no port, the value of port will be -1, which will be lead error.

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**